src/main/webapp/app/overview/course-overview.service.ts

```diff
@@ -45,13 +46,17 @@ export class CourseOverviewService {
         }
     }
 
-    getCorrespondingExerciseGroupByDate(date: dayjs.Dayjs | undefined): TimeGroupCategory {
-        if (!date) {
+    getCorrespondingExerciseGroupByDate(exercise: Exercise): TimeGroupCategory {
+        const start = exercise.startDate ?? exercise.releaseDate;
```

Comment:
Can you move this to `startDate` or even the check for current exercises?

Reply:
done